server/src/main/java/org/opentosca/toscana/plugins/cloudformation/mapper/CapabilityMapper.java

```diff
@@ -0,0 +1,310 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.opentosca.toscana.model.capability.ComputeCapability;
+import org.opentosca.toscana.model.capability.OsCapability;
+import org.opentosca.toscana.plugins.util.TransformationFailureException;
+
+import com.amazonaws.SdkClientException;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.ec2.AmazonEC2;
+import com.amazonaws.services.ec2.AmazonEC2ClientBuilder;
+import com.amazonaws.services.ec2.model.DescribeImagesRequest;
+import com.amazonaws.services.ec2.model.DescribeImagesResult;
+import com.amazonaws.services.ec2.model.Filter;
+import com.amazonaws.services.ec2.model.Image;
+import com.google.common.collect.ImmutableList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CapabilityMapper {
+
+    private final static Logger logger = LoggerFactory.getLogger(CapabilityMapper.class);
+
+    private final ImmutableList<InstanceType> EC2_INSTANCE_TYPES = ImmutableList.<InstanceType>builder()
+        .add(new InstanceType("t2.nano", 1, 512))
+        .add(new InstanceType("t2.micro", 1, 1024))
+        .add(new InstanceType("t2.small", 1, 2048))
+        .add(new InstanceType("t2.medium", 2, 4096))
+        .add(new InstanceType("t2.large", 2, 8192))
+        .add(new InstanceType("t2.xlarge", 4, 16384))
+        .add(new InstanceType("t2.2xlarge", 8, 32768))
+        .build();
+
+    private final ImmutableList<InstanceType> RDS_INSTANCE_CLASSES = ImmutableList.<InstanceType>builder()
+        .add(new InstanceType("db.t2.micro", 1, 1024))
+        .add(new InstanceType("db.t2.small", 1, 2048))
+        .add(new InstanceType("db.t2.medium", 2, 4096))
+        .add(new InstanceType("db.t2.large", 2, 8192))
+        .add(new InstanceType("db.t2.xlarge", 4, 16384))
+        .add(new InstanceType("db.t2.2xlarge", 8, 32768))
+        .add(new InstanceType("db.m4.4xlarge", 16, 65536))
+        .add(new InstanceType("db.m4.10xlarge", 40, 163840))
+        .add(new InstanceType("db.m4.16xlarge", 64, 262144))
+        .build();
+
+    /**
+     This method requests the AWS server for ImageIds with filters which are filled based on
+     the values of the OsCapability. The image with the latest creation date is picked and its imageId returned.
+
+     @param osCapability The OsCapability to map.
+     @return A String that contains a valid ImageId that can be added to the properties of an ec2.
+     */
+    public String mapOsCapabilityToImageId(OsCapability osCapability) {
+        AmazonEC2 ec2 = AmazonEC2ClientBuilder.standard()
+            .withRegion(Regions.US_WEST_2) //TODO get this from user
+            .build();
+        //need to set these, owner are self and amazon
+        DescribeImagesRequest describeImagesRequest = new DescribeImagesRequest()
+            .withFilters(
+                new Filter("virtualization-type").withValues("hvm"),
```

Comment:
Are these always the same or do we have to set them dynamically in the future?
Same for most of the other hard-coded Strings in the Filters.